src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MongoScheduler.java

```diff
@@ -1,73 +1,55 @@
 package uk.nhs.digital.nhsconnect.nhais.mesh;
 
-import com.mongodb.client.result.UpdateResult;
+import java.time.LocalDateTime;
+
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.bson.Document;
+import uk.nhs.digital.nhsconnect.nhais.repository.SchedulerTimestampRepositoryExtensions;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.data.mongodb.core.MongoOperations;
-import org.springframework.data.mongodb.core.query.Update;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
-import java.time.LocalDateTime;
-
-import static org.springframework.data.mongodb.core.query.Criteria.where;
-import static org.springframework.data.mongodb.core.query.Query.query;
-
 @Component
 @Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class MongoScheduler {
 
-    @Autowired
-    private MongoOperations mongoOperations;
+    private final MeshClient meshClient;
+    private final SchedulerTimestampRepositoryExtensions schedulerTimestampRepository;
 
-    @Autowired
-    private MeshClient meshClient;
+    @Value("${nhais.scheduler.interval}")
+    private long minutes;
 
-    private static final String MESH_TIMESTAMP_COLLECTION_NAME = "mesh_timestamp";
-    private static final String TIMESTAMP_FIELD_NAME = "timestamp";
+    private static final String SCHEDULER_TYPE = "meshTimestamp";
+    private static final String MESH_TIMESTAMP_COLLECTION_NAME = "schedulerTimestamp";
 
     @Scheduled(fixedRate = 6000)
+    //TODO: set to 60_000
     public void updateConditionally() {
         LOGGER.debug("Scheduled job for mesh messages fetching started");
         if (updateTimestamp()) {
+            LOGGER.debug("Timestamp in {} collection has been set to current timestamp.", MESH_TIMESTAMP_COLLECTION_NAME);
             LOGGER.info("Mesh messages fetching started");
+
             for (String messageId : meshClient.getInboxMessageIds()) {
                 meshClient.getEdifactMessage(messageId);
             }
         } else {
+            LOGGER.debug("Timestamp in {} collection is after five minutes ago, so it has not been modified", MESH_TIMESTAMP_COLLECTION_NAME);
```

Comment:
When it's after 5 minutes then we update it, so I suppose it should say `is before`

Reply:
Also time is configurable, so time value should be dynamic

Reply:
no, it should be after. When it is 13.40, then five minutes ago is 13.35, so i.e. 13.37 will be after that 'five minutes ago' == 13.35

Reply:
configurable time added

Reply:
..."Timestamp in {} collection less than {} minutes in the past, so it has not been modified", MESH_TIMESTAMP_COLLECTION_NAME, minutes);